indico/htdocs/js/indico/jquery/categorypicker.js

```diff
@@ -0,0 +1,332 @@
+        },
+
+        _createInDialog: function() {
+            var self = this;
+            ajaxDialog({
+                title: self.options.dialogTitle,
+                content: $('<div>', {class: 'category-picker-wrapper'}).append($('<div>', {class: 'js-test'})),
+                onOpen: function(dialog) {
+                    self.element = dialog.contentContainer.find('.js-test');
+                    self.dialog = dialog;
+                    self._createInline();
+                }
+            });
+        },
+
+        _createNavigator: function() {
+            var self = this;
+            self.$category = $('<div>');
+            self.$categoryTree = $('<ul>', {class: 'group-list fixed-height'});
+            self.$categoryResults = $('<ul>', {class: 'group-list fixed-height'});
+            self.$categoryNavigator = $('<div>', {class: 'category-navigator i-box just-group-list with-hover-effect'})
+                .append($('<div>', {class: 'i-box-content'})
+                    .append(self.$category)
+                    .append(self.$categoryTree)
+                    .append(self.$categoryResults));
+            self.element.append(self.$categoryNavigator);
+
+            if (self.options.height) {
+                self.$category.closest('.category-picker').height(self.options.height);
+            }
+        },
+
+        _createSearchField: function() {
+            var self = this;
+
+            // Visible search field
+            self.$searchInput = $('<input>', {
+                class: 'js-search-category js-typeahead',
+                type: 'search',
+                name: 'q',
+                placeholder: $T.gettext("Search")
+            }).attr('autocomplete', 'off');
+
+            // Insert elements in DOM
+            var $typeaheadForm = $($.parseHTML(self._typeaheadTemplate)[0]);
+            $typeaheadForm.find('.typeahead__query').append(self.$searchInput);
+            self.element.prepend($typeaheadForm);
+
+            // Typeahead init
+            self.$searchInput.typeahead({
+                dynamic: true,
+                filter: false,
+                source: {
+                    ajax: {
+                        url: build_url(Indico.Urls.Categories.search),
+                        path: 'categories',
+                        data: {
+                            q: "{{query}}"
+                        }
+                    }
+                },
+                resultContainer: false,
+                cancelButton: false,
+                display: 'title',
+                callback: {
+                    onClickAfter: function(node, a, item) {
+                        self.goToCategory(item.id);
+                    },
+                    onResult: function(node, query, result) {
+                        self._renderResults(result);
+                    },
+                    onHideLayout: function() {
+                        self.$categoryResults.html('');
```

Comment:
It makes it clear you are dealing with jquery objects. Makes my life less miserable when building HTML in JS.